sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -662,40 +575,25 @@ async Task<Page<IncomingRelationship>> NextPageFunc(string nextLink, int? pageSi
         /// See the asynchronous version of this method for examples.
         /// </seealso>
         public virtual Pageable<IncomingRelationship> GetIncomingRelationships(string digitalTwinId, CancellationToken cancellationToken = default)
+            => PageableHelpers.CreateEnumerable(GetIncomingRelationshipsImplAsync(false, digitalTwinId, cancellationToken).EnsureSyncEnumerable(),
+                _clientDiagnostics, "DigitalTwinsGeneratedClient.ListIncomingRelationships");
+
+        private async IAsyncEnumerable<Page<IncomingRelationship>> GetIncomingRelationshipsImplAsync(bool async, string digitalTwinId, [EnumeratorCancellation] CancellationToken cancellationToken)
```

Comment:
Creating `Pageable` and `AsyncPageable` can be created from the same method that handles both `sync` and `async` code paths, which reduces the amount of copy/paste.

Reply:
I'm not sure about merging the method into one. Seems like most size reduction is coming from removing scoping logic.